docs/changehistory/NextVersion.md

```diff
@@ -657,763 +432,688 @@ The signatures to several methods in [BriefcaseManager]($backend) and [Briefcase
+      ),
+    },
+  });
+  ```
+
+#### `ControlledTree` API changes
+
+`ControlledTree` component has received the following breaking changes:
+
+- The component now takes `TreeModel` rather than `VisibleTreeNodes` as a prop to avoid requiring consumers to manage `VisibleTreeNodes` object. As a result, the `useVisibleTreeNodes` hook was replaced with `useTreeModel` hook. Typical migration:
+
+  **Before:**
+
+  ```tsx
+  const visibleNodes = useVisibleTreeNodes(modelSource);
+  return <ControlledTree visibleNodes={visibleNodes} {...otherProps} />;
+  ```
+
+  **After:**
+
+  ```tsx
+  const treeModel = useTreeModel(modelSource);
+  return <ControlledTree model={treeModel} {...otherProps} />;
+  ```
+
+- Name of the `treeEvents` prop was changed to `eventsHandler` to make it clearer. Typical migration:
+
+  **Before:**
+
+  ```tsx
+  return <ControlledTree treeEvents={eventsHandler} {...otherProps} />;
+  ```
+
+  **After:**
+
+  ```tsx
+  return <ControlledTree eventsHandler={eventsHandler} {...otherProps} />;
+  ```
+
+- `width` and `height` properties are now required. Previously they were optional and forced us to use non-optimal approach when not provided. Now it's up to the consumer to tell the size of the component. Typical migration:
+
+  **Before:**
+
+  ```tsx
+  return <ControlledTree {...props} />;
+  ```
+
+  **After:**
+
+  ```tsx
+  const width = 100;
+  const height = 100;
+  return <ControlledTree width={width} height={height} {...props} />;
+  ```
+
+  `width` and `height` props may be calculated dynamically using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) API.
+
+#### PropertyGrid - related API Changes
+
+- `width` and `height` are now required props for `VirtualizedPropertyGrid` and `VirtualizedPropertyGridWithDataProvider`. Also, `width` is now a required property for `PropertyList`. Previously they were optional and forced us to use non-optimal approach when not provided. Now it's up to the consumer to tell the size of the component. Typical migration:
+
+  **Before:**
+
+  ```tsx
+  return <VirtualizedPropertyGrid {...props} />;
+  ```
+
+  **After:**
+
+  ```tsx
+  const width = 100;
+  const height = 100;
+  return <VirtualizedPropertyGrid width={width} height={height} {...props} />;
+  ```
+
+  `width` and `height` props may be calculated dynamically using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) API.
 
 - Default value of `PresentationPropertyDataProvider.isNestedPropertyCategoryGroupingEnabled` was changed from `false` to `true`.
 
-### Deprecated Components in Favor of iTwinUI-react Components
+It is no longer necessary to supply a [Viewport]($frontend) when creating a [GraphicBuilder]($frontend). Instead, you can supply to [RenderSystem.createGraphic]($frontend) a [CustomGraphicBuilderOptions]($frontend) containing a function that can compute the level of detail appropriate for the produced [RenderGraphic]($frontend).
```

Comment:
@pmconne, this paragraph doesn't belong here. Do you know where to move it?

Reply:
Half of that paragraph disappeared some time ago. Restored and moved.